src/tribler/core/upgrade/upgrade.py

```diff
@@ -68,6 +70,22 @@ def cleanup_noncompliant_channel_torrents(state_dir):
                                  file_path)
 
 
+def catch_db_is_corrupted_exception(upgrader_method):
+
+    @wraps(upgrader_method)
+    def new_method(*args, **kwargs):
+        try:
+            upgrader_method(*args, **kwargs)
+        except DatabaseIsCorrupted as exc:
+            self: TriblerUpgrader = args[0]
+            self._logger.exception(exc)
+
+            if not self._db_is_corrupted_exception:
+                self._db_is_corrupted_exception = exc
+
+    return new_method
```

Comment:
It is relatively easy to add tests and satisfy Codecov. I can help you with them if you want.

Reply:
Done